Wire Create Domains Flow

rdig opened the pull request:
## Description

This PR wires in the domain create action in the UAC flow. This presents a couple of challenges:
- adding the action itself
- uploading metadata to ipfs
- reading that ipfs metadata in the action
- adding annotations
- wiring in the subgraph action itself

Note that this uses JoinColony/subgraph#10 to update the subgraph part, so you'll need to run `npm i` and `npm provision` in order to make this work

**Changes**

- [x] Added Redux Domain Create Actions
- [x] Removed old Redux Domain Create 
- [x] Add `createDomainAction ` saga
- [x] Refactored `CreateEditDialog` to properly send data and clamp the inputs (based on product input)
- [x] Added subgraph domain queries and resolvers
- [x] Fetch all domains from the subpgraph and populate them in the app
- [x] Create domain action copy
- [x] Updated `DetailsWidget` to show domain details
- [x] Refactored `ActionsPage` to support a fallback domain
- [x] Removed `ColonyActions` constant polling
- [x] Renamed  instaces of domains -> teams
- [x] Added domain description display in the sidebar
- [x] Refactored `DomainDropdown` to properly show domains
- [x] Refactored `DomainDropdown` to sort domains properly
- [x] Refactored `DomainDropdown` to be able to open create domain modals
- [x] Removed all `admin` module domain creation

**Demo**

![demo-create-domain-action](https://user-images.githubusercontent.com/1193222/104972142-8fd8a200-59f9-11eb-9cd2-fd363d805c21.gif)

![demo-domains-sidebar](https://user-images.githubusercontent.com/1193222/104972147-91a26580-59f9-11eb-9384-73e9cdc23d47.gif)

Resolves DEV-68
Resolves DEV-58
Resolves DEV-146
Resolves DEV-145
Resolves DEV-138

ArmandoGraterol commented on src/modules/dashboard/components/ManageDomainsDialog/ManageDomainsDialog.tsx:
```diff
@@ -13,24 +13,24 @@ import { canArchitect } from '../../../users/checks';
     id: 'dashboard.ManageDomainsDialog.createNewDomainDescription',
     defaultMessage:
-      'Teams, departments, circles: domains let you froup types of activity.',
+      'Domains, departments, circles: teams let you froup types of activity.',
```
Typo here: `froup`

ArmandoGraterol reviewed (changes requested):
After reading this chunky PR, I have become enlightened :lotus_position_man: 

I only noticed some small things. Tested it and it working great for me :ok_hand:

rdig commented on src/data/graphql/typeDefs.ts:
```diff
@@ -123,9 +123,14 @@ export default gql`
   extend type Domain {
-    # TODO guarantee color (resolver will always return a color)
     color: Int
     description: String
```
Yes. Good catch

rdig commented:
Fixed the last review issues, rebased (plus added a couple of small fixes I've noticed after the rebase -- actions list sorting was not working)

So @JoinColony/frontend I'd appreciate another review :)

ArmandoGraterol reviewed (approved):
Tested it again, still working as expected :ok_hand: It's lit
